cmd/korg/audit.go

```diff
@@ -0,0 +1,336 @@
+	for _, exceptionalUser := range exceptionalUsers {
+		if exceptionalUser == username {
+			return true
+		}
+	}
+
+	return false
+}
+
+func OrgAudit(o Options) error {
+	fmt.Printf("Running analysis with a lookback period of %s and activity threshold of %d\n", o.Period, o.ActivityThreshold)
+
+	var exceptionalUsers []string
+	if o.ExceptionsFile != "" {
+		fmt.Printf("reading exceptions from %s\n", o.ExceptionsFile)
+		exceptions, err := ReadExceptions(o.ExceptionsFile)
+		if err != nil {
+			return err
+		}
+
+		// build indexable map for exceptions
+		for _, exception := range exceptions {
+			exceptionalUsers = append(exceptionalUsers, exception.Username)
+		}
+
+		// Print exceptions to stdout
+		fmt.Println("Total Exceptions:", len(exceptions))
+		table := tablewriter.NewWriter(os.Stdout)
+		table.SetHeader([]string{"Username", "Reason"})
+		for _, exception := range exceptions {
+			table.Append([]string{exception.Username, exception.Reason})
+		}
+		table.Render()
+	}
+
+	fmt.Println("fetching data from devstats")
+	contributions, err := GetContributionsForAYear(o.Period)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("fetching org members")
+
+	// TODO(palnabarun): Get userinfo along with team memberships
+	userOrgs, err := GetAllUsersInOrgs(o, validOrgs)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println("filtering org members")
+	var orgMembersBelowThresholdAfterException []UserInfo
+	for username, orgs := range userOrgs {
+		if usernameInExceptions(exceptionalUsers, username) {
+			fmt.Printf("username %s in exceptions. skipping...\n", username)
+			continue
+		}
+		if usernameNotInContributors(contributions, username) ||
+			usernameBelowActivityThreshold(contributions, username, o.ActivityThreshold) {
+			orgMembersBelowThresholdAfterException = append(orgMembersBelowThresholdAfterException,
+				UserInfo{
+					Username:      username,
+					Orgs:          orgs,
+					Contributions: contributions[username].ContribCount,
+				})
+		}
+	}
+
+	// populate if member is an owner
+	if o.CheckOwners {
+		for i, member := range orgMembersBelowThresholdAfterException {
+			isOwner, err := IsOwner(member.Username)
+			if err != nil {
+				return err
+			}
+			fmt.Printf("checking if user %s is owner: %v\n", member.Username, isOwner)
+			member.IsOwner = isOwner
+			orgMembersBelowThresholdAfterException[i] = member
+		}
+	}
+
+	fmt.Println("Total \"Org Members\":", len(userOrgs))
+	fmt.Println("Total \"Org Members\" below threshold after exceptions:", len(orgMembersBelowThresholdAfterException))
+
+	switch o.Output {
+	case "table":
+		table := tablewriter.NewWriter(os.Stdout)
+		table.SetHeader([]string{"Username", "Orgs", "Owner?", "Owners Link"})
+
+		for _, v := range orgMembersBelowThresholdAfterException {
+
+			table.Append([]string{
+				v.Username,
+				strings.Join(v.Orgs, ", "),
+				fmt.Sprintf("%v", v.IsOwner),
+				fmt.Sprintf("https://go.k8s.io/owners/%s", v.Username),
+			})
```

Comment:
@palnabarun, thanks a lot! It looks great to me.

Just a small suggestion — Can we render the owner URL only when the username _is_ actually an owner? It's just to make the owner usernames stand out visually in the table. 

That's all. Thank you! :slightly_smiling_face:  

```suggestion
                        row := []string{
                                v.Username,
                                strings.Join(v.Orgs, ", "),
                                fmt.Sprintf("%v", v.IsOwner),
                        }

                        if v.IsOwner {
                                row = append(row, fmt.Sprintf("https://go.k8s.io/owners/%s", v.Username))
                        } else {
                                row = append(row, "") // Add an empty string as a placeholder
                        }

                        table.Append(row)
```